src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/DotNetXamarinProject.cs

```diff
@@ -113,30 +114,33 @@ public virtual ProjectRootElement Construct ()
 
 		public override string SaveProject ()
 		{
+			XNamespace ns = "http://schemas.microsoft.com/developer/msbuild/2003";
+			var encoding = Encoding.UTF8;
 			var root = Construct ();
-			var sw = new StringWriter ();
-			root.Save (sw);
-			var document = XDocument.Parse (sw.ToString ());
-			var pn = XName.Get ("Project", "http://schemas.microsoft.com/developer/msbuild/2003");
-			var p = document.Element (pn);
-			if (p != null) {
-				//NOTE: when running tests inside VS 2019 "Current" was set here
-				p.SetAttributeValue ("ToolsVersion", "15.0");
-
-				var referenceGroup = p.Elements ().FirstOrDefault (x => x.Name.LocalName == "ItemGroup" &&  x.HasElements && x.Elements ().Any (e => e.Name.LocalName == "Reference"));
-				if (referenceGroup != null) {
-					foreach (var pr in PackageReferences) {
-						//NOTE: without the namespace it puts xmlns=""
-						XNamespace ns = "http://schemas.microsoft.com/developer/msbuild/2003";
-						var e = XElement.Parse ($"<PackageReference Include=\"{pr.Id}\" Version=\"{pr.Version}\"/>");
-						e.Name = ns + e.Name.LocalName;
-						referenceGroup.Add (e);
+			XDocument document;
+			using (var stream = new MemoryStream ())
+			using (var sw = new StreamWriter (stream, encoding, 8 * 1024, leaveOpen: true)) {
+				root.Save (sw);
+				stream.Position = 0;
+				document = XDocument.Load (stream);
```

Comment:
any reason why we could not use the XDeclaration.Encoding property? https://docs.microsoft.com/en-us/dotnet/api/system.xml.linq.xdeclaration.encoding?view=netframework-4.8

Reply:
I found a way to set it:
```csharp
document.Declaration.Encoding = encoding.HeaderName;
```
The value was already `utf-8`, but I think it's probably better to do this anyway.